File: app/src/main/java/jp/hazuki/yuzubrowser/userjs/UserScriptEditActivity.java
package jp.hazuki.yuzubrowser.userjs;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.MenuItem.OnMenuItemClickListener;
import android.widget.EditText;

import jp.hazuki.yuzubrowser.R;
import jp.hazuki.yuzubrowser.utils.app.ThemeActivity;

public class UserScriptEditActivity extends ThemeActivity {
    public static final String EXTRA_USERSCRIPT = "UserScriptEditActivity.extra.userscript";

    private UserScriptInfo mUserScript;
    private EditText editText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.scroll_edittext);
        editText = (EditText) findViewById(R.id.editText);

        Intent intent = getIntent();
        if (intent == null)
            throw new NullPointerException("intent is null");
        mUserScript = intent.getParcelableExtra(EXTRA_USERSCRIPT);
        if (mUserScript == null) {
            mUserScript = new UserScriptInfo();
        } else {
            setTitle(new UserScript(mUserScript).getName());
            editText.setText(mUserScript.getData());
        }
    }

    @Override
    public void onBackPressed() {
        showSaveDialog(true);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        menu.add(R.string.userjs_save).setOnMenuItemClickListener(new OnMenuItemClickListener() {
            @Override
            public boolean onMenuItemClick(MenuItem arg0) {
                showSaveDialog(false);
                return false;
            }
        });
        return super.onCreateOptionsMenu(menu);
    }

    public void showSaveDialog(final boolean orFinish) {
        AlertDialog.Builder builder = new AlertDialog.Builder(this)
                .setTitle(R.string.confirm)
                .setMessage(R.string.userjs_save_confirm)
                .setPositiveButton(R.string.save, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface arg0, int arg1) {
                        mUserScript.setData(editText.getText().toString());
                        Intent intent = new Intent();
                        intent.putExtra(EXTRA_USERSCRIPT, mUserScript);
                        setResult(RESULT_OK, intent);
                        finish();
                    }
                })
                .setNeutralButton(android.R.string.cancel, null);
        if (orFinish)
            builder.setNegativeButton(R.string.not_save, new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialog, int which) {
                    finish();
                }
            });
        builder.show();
    }
}

File: legacy/src/main/java/jp/hazuki/yuzubrowser/legacy/userjs/UserScriptEditActivity.kt
/*
 * Copyright (C) 2017-2019 Hazuki
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jp.hazuki.yuzubrowser.legacy.userjs

import android.app.AlertDialog
import android.content.Intent
import android.os.Bundle
import android.view.Menu

import jp.hazuki.yuzubrowser.legacy.R
import jp.hazuki.yuzubrowser.ui.app.ThemeActivity
import kotlinx.android.synthetic.main.scroll_edittext.*

class UserScriptEditActivity : ThemeActivity() {

    private lateinit var mUserScript: UserScriptInfo

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.scroll_edittext)

        val intent = intent ?: throw NullPointerException("intent is null")
        val id = intent.getLongExtra(EXTRA_USERSCRIPT, -1)
        mUserScript = UserScriptDatabase.getInstance(applicationContext)[id] ?: UserScriptInfo()

        title = intent.getStringExtra(Intent.EXTRA_TITLE) ?: ""
        editText.setText(mUserScript.data)
    }

    override fun onBackPressed() {
        showSaveDialog(true)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menu.add(R.string.userjs_save).setOnMenuItemClickListener {
            showSaveDialog(false)
            false
        }
        return super.onCreateOptionsMenu(menu)
    }

    private fun showSaveDialog(orFinish: Boolean) {
        val builder = AlertDialog.Builder(this)
                .setTitle(R.string.confirm)
                .setMessage(R.string.userjs_save_confirm)
                .setPositiveButton(R.string.save) { _, _ ->
                    mUserScript.data = editText.text.toString()
                    UserScriptDatabase.getInstance(applicationContext).set(mUserScript)
                    setResult(RESULT_OK)
                    finish()
                }
                .setNeutralButton(android.R.string.cancel, null)
        if (orFinish)
            builder.setNegativeButton(R.string.not_save) { _, _ -> finish() }
        builder.show()
    }

    companion object {
        const val EXTRA_USERSCRIPT = "UserScriptEditActivity.extra.userscript"
    }
}
